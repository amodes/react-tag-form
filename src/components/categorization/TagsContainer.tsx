import React from "react";

import { TagRow } from "./TagRow";
import { SupplierData } from "./interfaces";
import {
  suggestedTagsCertifications,
  suggestedTagsPortfolio,
} from "./dummyData";

interface Props {
  supplierData: SupplierData | null;
  deleteTag: (type: keyof SupplierData, index: number) => void;
  addTag: (type: keyof SupplierData, value: string) => void;
}

export const TagsContainer = ({ supplierData, deleteTag, addTag }: Props) => {
  return (
    <>
      <TagRow
        title="General"
        tags={supplierData?.["tags-general"].map((entry) => entry.name) || []}
        type="tags-general"
        deleteTag={deleteTag}
        addTag={addTag}
      />
      <TagRow
        title="Portfolio"
        tags={supplierData?.["tags-portfolio"].map((entry) => entry.name) || []}
        type="tags-portfolio"
        deleteTag={deleteTag}
        addTag={addTag}
        suggestions={suggestedTagsPortfolio}
      />
      <TagRow
        title="Certificates"
        tags={
          supplierData?.["tags-certificates"].map((entry) => entry.name) || []
        }
        type="tags-certificates"
        deleteTag={deleteTag}
        addTag={addTag}
        suggestions={suggestedTagsCertifications}
      />
    </>
  );
};
