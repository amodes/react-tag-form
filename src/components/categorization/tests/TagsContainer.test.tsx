import React from "react";
import renderer from "react-test-renderer";

import { TagsContainer } from "../TagsContainer";

describe("<TagsContainer />", () => {
  it("should render correctly", () => {
    const tree = renderer
      .create(
        <TagsContainer
          supplierData={{
            "tags-general": [
              { name: "Example general tag", id: 1, type: "tags-general" },
            ],
            "tags-portfolio": [],
            "tags-certificates": [
              {
                name: "Example certificate tag",
                id: 2,
                type: "tags-certificates",
              },
            ],
          }}
          deleteTag={() => {}}
          addTag={() => {}}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
