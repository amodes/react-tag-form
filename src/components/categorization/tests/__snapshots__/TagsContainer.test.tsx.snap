// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TagsContainer /> renders correctly 1`] = `
Array [
  <div
    className="sc-jSMfEi AlZcL"
  >
    <div
      className="sc-gKXOVf fvGQzg"
    >
      General
    </div>
    <div
      className="sc-iBkjds crHxek"
    >
      <div
        className="sc-bczRLJ ljJSEx"
      >
        Example general tag
      </div>
      <div
        className="sc-dkzDqf bhAngL"
      >
        <div
          className="sc-eCYdqJ zteIz"
          onClick={[Function]}
        >
          + New Tag
        </div>
      </div>
    </div>
  </div>,
  <div
    className="sc-jSMfEi AlZcL"
  >
    <div
      className="sc-gKXOVf fvGQzg"
    >
      Portfolio
    </div>
    <div
      className="sc-iBkjds crHxek"
    >
      <div
        className="sc-dkzDqf bhAngL"
      >
        <div
          className="sc-eCYdqJ zteIz"
          onClick={[Function]}
        >
          + New Tag
        </div>
      </div>
    </div>
  </div>,
  <div
    className="sc-jSMfEi AlZcL"
  >
    <div
      className="sc-gKXOVf fvGQzg"
    >
      Certificates
    </div>
    <div
      className="sc-iBkjds crHxek"
    >
      <div
        className="sc-bczRLJ ljJSEx"
      >
        Example certificate tag
      </div>
      <div
        className="sc-dkzDqf bhAngL"
      >
        <div
          className="sc-eCYdqJ zteIz"
          onClick={[Function]}
        >
          + New Tag
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`<TagsContainer /> should render correctly 1`] = `
Array [
  <div
    className="sc-jSMfEi AlZcL"
  >
    <div
      className="sc-gKXOVf fvGQzg"
    >
      General
    </div>
    <div
      className="sc-iBkjds crHxek"
    >
      <div
        className="sc-bczRLJ ljJSEx"
      >
        Example general tag
      </div>
      <div
        className="sc-dkzDqf bhAngL"
      >
        <div
          className="sc-eCYdqJ zteIz"
          onClick={[Function]}
        >
          + New Tag
        </div>
      </div>
    </div>
  </div>,
  <div
    className="sc-jSMfEi AlZcL"
  >
    <div
      className="sc-gKXOVf fvGQzg"
    >
      Portfolio
    </div>
    <div
      className="sc-iBkjds crHxek"
    >
      <div
        className="sc-dkzDqf bhAngL"
      >
        <div
          className="sc-eCYdqJ zteIz"
          onClick={[Function]}
        >
          + New Tag
        </div>
      </div>
    </div>
  </div>,
  <div
    className="sc-jSMfEi AlZcL"
  >
    <div
      className="sc-gKXOVf fvGQzg"
    >
      Certificates
    </div>
    <div
      className="sc-iBkjds crHxek"
    >
      <div
        className="sc-bczRLJ ljJSEx"
      >
        Example certificate tag
      </div>
      <div
        className="sc-dkzDqf bhAngL"
      >
        <div
          className="sc-eCYdqJ zteIz"
          onClick={[Function]}
        >
          + New Tag
        </div>
      </div>
    </div>
  </div>,
]
`;
